WorldMoodTracker/js/components/WorldMap.js

```diff
@@ -13,12 +13,49 @@ export default class WorldMap extends React.Component {
         this.state = {
             map: null
         };
-        this.resize = this.resize.bind(this, this.state.map);
-        this.globalInfo = this.globalInfo.bind(this, this.state.global);
+        this.resize = this.resize.bind(this);
+        this.globalInfo = this.globalInfo.bind(this);
+        this.globalFetch = this.globalFetch.bind(this);
         // Resize map when window changes
         window.addEventListener('resize', this.resize);
     }
 
+    globalFetch() {
+        let since;
+        let now = new Date();
+        if (!this.state.global) {
+            since = '';
+        } else if (this.state.global.time === 'All') {
+            since = '';
+        } else if (this.state.global.time === 'Year') {
+            since = '&since=' + now.getFullYear();
+        } else if (this.state.global.time === 'Month') {
+            since = '&since=' + now.getFullYear() + '-' + now.getMonth();
+        } else if (this.state.global.time === 'Day') {
+            since = '&since=' + now.getFullYear() + '-' + now.getMonth() + '-' + now.getDate();
+        } else if (this.state.global.time === 'Hour') {
+            since = '&since=' + now.getFullYear() + '-' + now.getMonth() + '-' + now.getDate() + 'T' + now.getHours();
+        }
+        this.setState(...this.state, {global: {fetching: true, fetched: false, errored: false}, mapBorder: '#ffffff'});
+        fetchJsonp(host + '/api/classifier.json?minified=true&classifier=emotion&all=true' + since, {
```

Comment:
@vibhcool: I think that this is by far the only request that is to be made. We do not need many configurations here and hence the string should do.

Also, it will add some extra lines of code as the `since` parameter is optional.